src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -100,8 +100,10 @@ private async Task AddProjectsToSolutionAsync(string solutionFileFullPath, IEnum
             foreach (var projectPath in projectPaths)
             {
                 string relativePath = Path.GetRelativePath(Path.GetDirectoryName(solutionFileFullPath), projectPath);
-                // Add fallback solution folder
-                string relativeSolutionFolder = Path.GetDirectoryName(relativePath);
+                // Add fallback solution folder if relative path does not contain `..`.
+                string relativeSolutionFolder =  relativePath.Contains("..")
```

Comment:
I don't think Contains is correct. A normal valid path might contain a ".." for example "Legen......dary"

Reply:
You are absolutely right, replaced with splitting the path and checking if any of the subfolders _is_ `..`

Also nice reference :)